src/sql/parts/notebook/notebookActions.ts

```diff
@@ -48,6 +48,36 @@ export class AddCellAction extends Action {
 	}
 }
 
+export class SaveNotebookAction extends Action {
+	private static readonly notebookSavedMsg = localize('notebookSavedMsg', 'Notebook saved successfully.');
+	private static readonly notebookFailedSaveMsg = localize('notebookFailedSaveMsg', 'Failed to save Notebook.');
+	constructor(
+		id: string, label: string, cssClass: string,
+		@INotificationService private _notificationService: INotificationService
+	) {
+		super(id, label, cssClass);
+	}
+
+	public run(context: NotebookComponent): TPromise<boolean> {
```

Comment:
We need to fix this in the future. We shouldn't be using NotebookComponent, we should have a clean context object that only contains things that can be serialized across to extension-based buttons/actions.

Reply:
For sure